front/lib/models/assistant/actions/retrieval.ts

```diff
@@ -0,0 +1,214 @@
+    sequelize: front_sequelize,
+    hooks: {
+      beforeValidate: (retrieval: AssistantAgentRetrievalConfiguration) => {
+        // Validation for templated Query
+        if (retrieval.query == "templated") {
+          if (retrieval.queryTemplate === null) {
+            throw new Error("Must set a template for templated query");
+          }
+        } else {
+          if (retrieval.queryTemplate) {
+            throw new Error("Can't set a template without templated query");
+          }
+        }
+
+        // Validation for Timeframe
+        if (retrieval.relativeTimeFrame == "custom") {
+          if (
+            retrieval.relativeTimeFrameDuration === null ||
+            retrieval.relativeTimeFrameUnit === null
+          ) {
+            throw new Error(
+              "Custom relative time frame must have a duration and unit set"
+            );
+          }
+        }
+      },
+    },
+  }
+);
+
+/**
+ * Configuration of Datasources used for Retrieval Action.
+ */
+export class AssistantAgentDatasourceConfiguration extends Model<
+  InferAttributes<AssistantAgentDatasourceConfiguration>,
+  InferCreationAttributes<AssistantAgentDatasourceConfiguration>
+> {
+  declare id: number;
+
+  declare minTimestamp: number | null;
+  declare maxTimestamp: number | null;
+  declare timeframeDuration: number | null;
+  declare timeframeUnit: TimeframeUnit | null;
+
+  declare tagsIn: string[];
+  declare tagsOut: string[];
+  declare parentsIn: string[];
+  declare parentsOut: string[];
+
+  declare datasourceId: ForeignKey<DataSource["id"]>;
+  declare retrievalConfigurationId: ForeignKey<
+    AssistantAgentRetrievalConfiguration["id"]
+  >;
+}
+AssistantAgentDatasourceConfiguration.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    minTimestamp: {
+      type: DataTypes.INTEGER,
+      allowNull: true,
+    },
+    maxTimestamp: {
+      type: DataTypes.INTEGER,
+      allowNull: true,
+    },
+    timeframeDuration: {
+      type: DataTypes.INTEGER,
+      allowNull: true,
+    },
+    timeframeUnit: {
+      type: DataTypes.STRING,
+      allowNull: true,
+    },
+    tagsIn: {
+      type: DataTypes.ARRAY(DataTypes.STRING),
+      allowNull: false,
+      defaultValue: [],
+    },
+    tagsOut: {
+      type: DataTypes.ARRAY(DataTypes.STRING),
+      allowNull: false,
+      defaultValue: [],
+    },
+    parentsIn: {
+      type: DataTypes.ARRAY(DataTypes.STRING),
+      allowNull: false,
+      defaultValue: [],
+    },
+    parentsOut: {
+      type: DataTypes.ARRAY(DataTypes.STRING),
+      allowNull: false,
+      defaultValue: [],
+    },
+  },
+  {
+    modelName: "assistant_agent_datasource_configuration",
+    sequelize: front_sequelize,
+    hooks: {
+      beforeValidate: (
+        datasourceConfig: AssistantAgentDatasourceConfiguration
+      ) => {
+        if (
+          (datasourceConfig.minTimestamp === null) !==
```

Comment:
same here for the null (sorry my fault but realized when testing my hook earlier that fields are never null)